core/modules/cart/helpers/productsEquals.ts

```diff
@@ -20,14 +15,73 @@ const isServerIdsEquals = (product1: CartItem, product2: CartItem): boolean => {
   return areItemIdsDefined && product1ItemId === product2ItemId
 }
 
+// 'checksum' check
+const getChecksum = (product: CartItem) => {
```

Comment:
```suggestion
const getChecksum = (product: CartItem) => product.checksum ? product.checksum : productChecksum(product)
```
mayby shorter form, wdyt?  :)